temba/classifiers/types/dialogflow/train_bot.py

```diff
@@ -0,0 +1,197 @@
+import csv
+import logging
+
+from google.cloud import dialogflow_v2
+
+
+logger = logging.getLogger(__name__)
+
+
+class TrainingClient:
+    def __init__(self, csv_data: list, languages: list, credential: dict):
+        self.credential = credential
+        self.csv_data = csv_data
+        self.languages = languages
+        fill_list = [[]] * len(languages)
+        fill_zero = [0] * len(languages)
+        errors_per_lang = dict(zip(languages, fill_list))
+        count_per_lang = dict(zip(languages, fill_zero))
+        self.messages = dict(
+            errors=[],
+            update_errors=errors_per_lang,
+            create_errors=errors_per_lang,
+            created=count_per_lang,
+            updated=count_per_lang,
+        )
+
+    def get_csv(self):
+        reader = csv.reader(self.csv_data)
+        data_list = list(reader)
+
+        header = data_list.pop(0)
+        return data_list, header
+
+    def csv_to_dict(self, language_code):
```

Comment:
Not sure if that would be the same concept, but there's a built-in function to read the CSV as a Dict: https://docs.python.org/3/library/csv.html#csv.DictReader